public/app/features/plugins/admin/state/selectors.ts

```diff
@@ -61,17 +61,7 @@ export const selectPlugins = (filters: PluginFilters) =>
         return false;
       }
 
-      if (filters.hasUpdate !== undefined && plugin.hasUpdate !== filters.hasUpdate) {
-        return false;
-      }
-
-      // plugins not controlled by the user should not be shown as updatable
-      if (
-        filters.hasUpdate !== undefined &&
-        filters.hasUpdate &&
-        plugin.hasUpdate &&
-        (plugin.isCore || plugin.isManaged || plugin.isProvisioned || plugin.isUpdatingFromInstance)
-      ) {
+      if (filters.hasUpdate !== undefined && (plugin.hasUpdate !== filters.hasUpdate || !isPluginUpdateable(plugin))) {
```

Comment:
this condition is a bit difficult to read 😅

Reply:
I struggled with this too at first. At least it's somewhat consistent with the ones above 😅